superset-frontend/src/explore/reducers/getInitialState.ts

```diff
@@ -52,40 +61,49 @@ export interface ExplorePageBootstrapData extends JsonObject {
 export default function getInitialState(
   bootstrapData: ExplorePageBootstrapData,
 ) {
-  const { form_data: initialFormData } = bootstrapData;
-  const { slice } = bootstrapData;
-  const sliceName = slice ? slice.slice_name : null;
+  const {
+    form_data: initialFormData,
+    common,
+    user,
+    datasource,
+    slice,
+  } = bootstrapData;
 
   const exploreState = {
     // note this will add `form_data` to state,
     // which will be manipulatable by future reducers.
-    ...bootstrapData,
-    sliceName,
-    common: {
-      flash_messages: bootstrapData.common.flash_messages,
-      conf: bootstrapData.common.conf,
-    },
+    can_add: findPermission('can_write', 'Chart', user?.roles),
+    can_download: findPermission('can_csv', 'Superset', user?.roles),
+    can_overwrite: ensureIsArray(slice?.owners).includes(
+      user?.userId as number,
```

Comment:
Shouldn't this be something like `user?.userId || 0` since the user can be `undefined`?

Reply:
Yeah that's the point - array of owners won't include `undefined` so user who's not logged in won't have `overwrite` permission. If we defaulted to 0, there's a small chance that there's an actual user with id 0 and we'd allow anonymous user to overwrite that user's charts 🙂